views/vm/show.erb

```diff
@@ -89,23 +89,55 @@
       </tbody>
     </table>
   </div>
-  <!-- Delete Card -->
-  <% if has_permission?("Vm:delete", @vm[:id]) %>
+  <!-- Danger Zone -->
+  <div>
+    <div class="md:flex md:items-center md:justify-between pb-2 lg:pb-4">
+      <div class="min-w-0 flex-1">
+        <h3 class="text-2xl font-bold leading-7 text-gray-900 sm:truncate sm:text-2xl sm:tracking-tight">
+          Danger Zone
+        </h3>
+      </div>
+    </div>
     <div class="overflow-hidden rounded-lg shadow ring-1 ring-black ring-opacity-5 bg-white divide-y divide-gray-200">
-      <div class="px-4 py-5 sm:p-6">
-        <div class="sm:flex sm:items-center sm:justify-between">
-          <div>
-            <h3 class="text-base font-semibold leading-6 text-gray-900">Delete virtual machine</h3>
-            <div class="mt-2 text-sm text-gray-500">
-              <p>This action will permanently delete this virtual machine. Deleted data cannot be recovered. Use it
-                carefully.</p>
+      <!-- Restart Card -->
+      <% if has_permission?("Vm:edit", @vm[:id]) %>
+        <div class="px-4 py-5 sm:p-6">
+          <form action="<%= "#{@project_data[:path]}#{@vm[:path]}/restart" %>" role="form" method="POST">
+            <%== csrf_tag("#{@project_data[:path]}#{@vm[:path]}/restart") %>
+            <div class="sm:flex sm:items-center sm:justify-between">
+              <div>
+                <h3 class="text-base font-semibold leading-6 text-gray-900">Restart virtual machine</h3>
+                <div class="mt-2 text-sm text-gray-500">
+                  <p>This action will restart the virtual machine. It will be offline momentarily, and all connections
```

Comment:
Something like this might work better for vms.
```
This action will restart the virtual machine, causing it to be temporarily offline.
```

Reply:
Thanks